src/framework/audio/internal/synthesizers/fluidsynth/fluidsequencer.cpp

```diff
@@ -130,8 +130,13 @@ void FluidSequencer::updatePlaybackEvents(EventSequenceMap& destination, const m
 
             destination[timestampTo].emplace(std::move(noteOff));
 
-            appendControlSwitch(destination, noteEvent, PEDAL_CC_SUPPORTED_TYPES, midi::SUSTAIN_PEDAL_CONTROLLER);
-            appendPitchBend(destination, noteEvent, BEND_SUPPORTED_TYPES, channelIdx);
+            for (const auto& pair : noteEvent.expressionCtx().articulations) {
+                if (muse::contains(PEDAL_CC_SUPPORTED_TYPES, pair.first)) {
```

Comment:
Apologies if it's a silly question, but is it possible for this conditional to execute more than once throughout the loop? If so, is it OK to call `appendControlSwitch` more than once for a given `noteEvent`?

Reply:
It should be ok to call appendControlSwitch more than once. In practice, that won't happen simply because we only use it for the pedal right now (and each noteEvent will have only 1 pedal articulation)